### PR TITLE
Add FAQ entry explaining Numba project name

### PR DESCRIPTION
**Pull request by cdeil**
This PR adds a FAQ entry explaining the Numba project name.

I'm giving a talk on Numba at a small meeting on Friday, and this was one of the things I was wondering for the introduction part. I did find this explanation for "Numba" mentioning Mamba in some YouTube video from ~ 2012, don't have the link at the moment.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Reference: https://youtu.be/WYi1cymszqY?t=596

File: docs/source/user/faq.rst [hunk 1]
```diff
@@ -272,6 +272,12 @@ value, for example::
 Miscellaneous
 =============
 
+What does the project name Numba mean?
+--------------------------------------
+
+"Numba" is a combination of "Number" or "Numpy" and "Mamba".
+Mambas are some of the fastest snakes in the world, and Numba makes you Python code fast.
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]:
```suggestion
Mambas are some of the fastest snakes in the world, and Numba makes your Python code fast.
```
also, this line need wrapping to 80 chars, it is too long.

File: docs/source/user/faq.rst [hunk 2]
```diff
@@ -272,6 +272,12 @@ value, for example::
 Miscellaneous
 =============
 
+What does the project name Numba mean?
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]: I think this is more "Where does the project name `Numba` from from?"

File: docs/source/user/faq.rst [hunk 3]
```diff
@@ -272,6 +272,12 @@ value, for example::
...
+--------------------------------------
+
+"Numba" is a combination of "Number" or "Numpy" and "Mamba".
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]:
```suggestion
"Numba" is a combination of "NumPy" and "Mamba".
```

**Comment by cdeil** [comment]
@stuartarchibald - Implemented your suggestions and force-pushed a new commit with this FAQ entry.

File: docs/source/user/faq.rst [hunk 4]
```diff
@@ -272,6 +272,12 @@ value, for example::
 Miscellaneous
 =============
 
+Where does the project name Numba come from?
+--------------------------------------------
+
+"Numba" is a combination of "Numpy" and "Mamba". Mambas are some of the fastest
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]:
```suggestion
"Numba" is a combination of "NumPy" and "Mamba". Mambas are some of the fastest
```

File: docs/source/user/faq.rst [hunk 5]
```diff
@@ -272,6 +272,12 @@ value, for example::
 Miscellaneous
 =============
 
+Where does the project name Numba come from?
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]:
```suggestion
Where does the project name "Numba" come from?
```

**Comment by cdeil** [comment]
@stuartarchibald - OK, will do.

In this case it's a docs entry with 3 lines, and I thought clean commit history is more important than re-review.

Is this a general rule to **always** add commits and never rewrite history once a PR is opened?
Do you squash small commits on merge?

Maybe add a few words of advice on this topic here?
http://numba.pydata.org/numba-doc/latest/developer/contributing.html

File: docs/source/user/faq.rst [hunk 6]
```diff
@@ -272,6 +272,12 @@ value, for example::
...
+--------------------------------------------
```
Comment (stuartarchibald) [inline docs/source/user/faq.rst]: This line of `'-'` will need extending WRT the above.

Comment (cdeil) [reply docs/source/user/faq.rst]: So I should accept your suggestions, then do a follow-up commit, creating 3 commits in total for this PR?

**Comment by stuartarchibald** [comment]
> @stuartarchibald - OK, will do.

Great, thanks.

> Is this a general rule to always add commits and never rewrite history once a PR is opened?
Do you squash small commits on merge?

If the git history is giant for a PR then sometimes it might be squashed. Keeping review history is v. important as it records thought process. It's also not particularly usual for people to force-push branches under review.

> Maybe add a few words of advice on this topic here?

Perhaps open a PR? IIRC It's not a regularly occurring problem.

File: docs/source/user/faq.rst (hunk 6 above)
Comment (stuartarchibald) [reply docs/source/user/faq.rst]: Up to you how you choose to do it. I, personally, would opt for just creating a single patch that makes the suggested changes and github will pick up on them.

**Comment by cdeil** [comment]
@stuartarchibald - Implemented your suggestions in 4c3d60cec89ba36de8edcd67dfd0ca6329044960 .

**Comment by seibert** [comment]
Merging even though some of the CI is still in progress because this is just a doc change.

**Pull request merged**